runtime/ast/type.go

```diff
@@ -64,6 +64,7 @@ type Type interface {
 	HasPosition
 	fmt.Stringer
 	isType()
+	Equal(other Type, checker TypeEqualityChecker) error
```

Comment:
Maybe rename this function to `CheckEqual`. `Equal` sounds like it just returns a boolean